app/settings/paymentSetup.js

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useState } from 'react'
+import { StripeProvider, useStripe } from '@stripe/stripe-react-native'
+import { Button, HelperText, Text } from 'react-native-paper'
+import { useDispatch, useSelector } from 'react-redux'
+import { Alert, StyleSheet } from 'react-native'
+
+import LoadingSpinner from '@components/LoadingSpinner'
+import Page from '@components/Page'
+import {
+    initPaymentSetup,
+    savePaymentMethod,
+    selectPaymentSetup,
+    selectPaymentLoading,
+    selectPaymentSetupError,
+    selectPaymentMethodSaved,
+    selectProfileData
+} from '@store'
+
+const PaymentSetup = () => {
+    const dispatch = useDispatch()
+    const { initPaymentSheet, presentPaymentSheet } = useStripe()
+
+    const [hasError, setHasError] = useState(false)
+    const [hasLoaded, setHasLoaded] = useState(false)
+
+    const profileData = useSelector(selectProfileData)
+    const paymentSetupIntent = profileData?.paymentSetupIntent
+
+    const paymentLoading = useSelector(selectPaymentLoading)
+    const paymentSetup = useSelector(selectPaymentSetup)
+    const paymentSetupError = useSelector(selectPaymentSetupError)
+    const paymentMethodSaved = useSelector(selectPaymentMethodSaved)
+
+    const initializePaymentSheet = async (paymentSetup) => {
+        const { paymentSetupIntent, ephemeralKey, customer } = paymentSetup
+
+        // TODO: add a returnURL
```

Comment:
did you mean to leave this in or when do you plan on coming back to this?

Reply:
It's intentional. We don't have a returnURL at this point in time.